synapse/handlers/relations.py

```diff
@@ -190,13 +190,11 @@ async def _get_bundled_aggregation_for_event(
         # while others need more processing during serialization.
         aggregations = BundledAggregations()
 
-        annotations = await self._main_store.get_aggregation_groups_for_event(
+        annotations, limited = await self._main_store.get_aggregation_groups_for_event(
             event_id, room_id
         )
-        if annotations.chunk:
-            aggregations.annotations = await annotations.to_dict(
-                cast("DataStore", self)
-            )
+        if annotations:
+            aggregations.annotations = {"chunk": annotations, "limited": limited}
```

Comment:
Previously this would have given pagination tokens instead of `limited`. I chose `limited` instead since I would think that a client wants *some* way to know if it got all the results or not. If it hasn't, then it should go check `/relations`. 🤷 (`limited` also now matches what the MSC says...)

Reply:
I'm not really following this. Are we no longer allowing pagination? Typically the absence of `next_batch`implies that we've reached the end of the results (see https://spec.matrix.org/v1.2/appendices/#pagination)

Which MSC says we should return `limited` ?

Isn't changing the format of the response to `/relations` orthogonal to removing `/aggregations` ?

Reply:
> I'm not really following this. Are we no longer allowing pagination? Typically the absence of `next_batch`implies that we've reached the end of the results (see https://spec.matrix.org/v1.2/appendices/#pagination)

Well `/aggregations` isn't really part of any MSC right now. It was removed from the (merged) MSC2675. MSC2677 defines the bundled aggregation format for `m.annotations`, but is mostly lies. The changes in here make it more closely follow MSC2677.

> Which MSC says we should return `limited` ?

MSC2677, see https://github.com/uhoreg/matrix-doc/blob/aggregations-reactions/proposals/2677-reactions.md#bundled-relations

It still wouldn't match perfectly though since we would not include the `count` key.

> Isn't changing the format of the response to `/relations` orthogonal to removing `/aggregations` ?

This doesn't change the response of `/relations`, it changes the bundled aggregations format. The idea is that you can still use `/relations` and aggregate locally even without `/aggregations` though. So the information isn't lost, just a bit harder to get to.

Let me know if that makes sense, or we can discuss offline if that's ab it easier!

Reply:
Based on our conversation in #synapse-dev:matrix.org I've removed this flag.